core/core/group_chain.go

```diff
@@ -0,0 +1,117 @@
+package xchaincore
+
+import (
+	"strings"
+	"sync"
+	"time"
+)
+
+type GroupChainRegister interface {
+	CheckGroupChain(bcname, remotePeerID string) bool
+	GetAllowedPeersWithBcname(bcname string) map[string]bool
+}
+
+type groupChainCache struct {
+	// key: peerID value: ip + peerID
+	StreamCache map[string]string
+	// key: bcname value: map[string]bool
+	StreamContractCache map[string]map[string]bool
+	// key: bcname value: bool
+	ChainContractCache map[string]bool
+	Mutex              *sync.Mutex
+}
+
+// CheckGroupChain 判断某条链下的某个节点是否在白名单中
+func (xm *XChainMG) CheckGroupChain(bcname, remotePeerID string) bool {
```

Comment:
I think 'FindNode' is more clear than 'CheckGroupChain'.

Reply:
Or 'FindPeer'

Reply:
Corrected with a new name of IsPeerInGroupChain